rpc/server.go

```diff
@@ -43,67 +42,71 @@ type Config struct {
 
 // NewServer ...
 func NewServer(cfg *Config) (*Server, error) {
```

Comment:
I pulled all the things that can fail in the lead-up to launching the server out of the `Start()` method and into this `NewServer` method. One of the things is allocating the listener, as that allows us to pass `0` for the port and have the OS assign us a free ephemeral port. This is useful when testing, when the actual port used is unimportant, but it must be a free port. We want to be able to query the allocated port back from the `Server` object before calling the blocking method `Start()`.